packages/gatsby/src/commands/build.js

```diff
@@ -201,4 +212,51 @@ module.exports = async function build(program: BuildArgs) {
   await stopTracer()
   workerPool.end()
   buildActivity.end()
+
+  if (
+    process.env.GATSBY_PAGE_BUILD_ON_DATA_CHANGES &&
+    process.argv.indexOf(`--log-pages`) > -1
+  ) {
+    if (pagePaths.length) {
+      report.info(
+        `Built pages:\n${pagePaths.map(
+          path => `Updated page: ${path}\n`
+        )}`.replace(/,/g, ``)
+      )
+    }
+    if (typeof deletedPageKeys !== `undefined` && deletedPageKeys.length) {
```

Comment:
Assuming `null` is not an important distinction, you can simplify this;

```suggestion
    if (deletedPageKeys?.length) {
```